tests/unit/backend/recipe/test_DiffractionCalibrationRecipe.py

```diff
@@ -25,87 +15,36 @@
 from snapred.backend.dao.state.InstrumentState import InstrumentState
 from snapred.backend.recipe.DiffractionCalibrationRecipe import DiffractionCalibrationRecipe as Recipe
 from snapred.meta.Config import Config, Resource
+from util.diffraction_calibration_synthetic_data import SyntheticData
+from util.helpers import deleteWorkspaceNoThrow
 
 ThisRecipe: str = "snapred.backend.recipe.DiffractionCalibrationRecipe"
 PixelCalAlgo: str = ThisRecipe + ".PixelDiffractionCalibration"
 GroupCalAlgo: str = ThisRecipe + ".GroupDiffractionCalibration"
 
 
-class TestDiffractionCalibtationRecipe(unittest.TestCase):
+class TestDiffractionCalibrationRecipe(unittest.TestCase):
     def setUp(self):
-        self.fakeRunNumber = "555"
-        fakeRunConfig = RunConfig(
-            runNumber=str(self.fakeRunNumber),
-            IPTS="",
-        )
-
-        fakeInstrumentState = InstrumentState.parse_raw(Resource.read("/inputs/calibration/sampleInstrumentState.json"))
-        fakeInstrumentState.particleBounds.tof.minimum = 1
-        fakeInstrumentState.particleBounds.tof.maximum = 10
-
-        fakeFocusGroup = FocusGroup.parse_raw(Resource.read("/inputs/diffcal/fakeFocusGroup.json"))
-        fakeFocusGroup.definition = Resource.getPath("inputs/testInstrument/fakeSNAPFocGroup_Natural.xml")
-
-        peakList = [
-            DetectorPeak.parse_obj({"position": {"value": 2, "minimum": 1, "maximum": 3}}),
-            DetectorPeak.parse_obj({"position": {"value": 5, "minimum": 4, "maximum": 6}}),
-        ]
-
-        self.fakeIngredients = TheseIngredients(
-            runConfig=fakeRunConfig,
-            focusGroup=fakeFocusGroup,
-            instrumentState=fakeInstrumentState,
-            groupedPeakLists=[
-                GroupPeakList(groupID=3, peaks=peakList, maxfwhm=0.01),
-                GroupPeakList(groupID=7, peaks=peakList, maxfwhm=0.02),
-                GroupPeakList(groupID=2, peaks=peakList, maxfwhm=0.02),
-                GroupPeakList(groupID=11, peaks=peakList, maxfwhm=0.02),
-            ],
-            convergenceThreshold=0.5,
-            calPath=Resource.getPath("outputs/calibration/"),
-            pixelGroup=fakeInstrumentState.pixelGroup,
-        )
+        self.syntheticInputs = SyntheticData()
```

Comment:
Should this be inside `setUpClass`?  Or is it better to re-run it before each test?

Reply:
**Tests may be run in parallel, although we generally do not do that.  I'm implementing all of my own tests as if this will be the case.**  We gain a _little_ bit of run time by caching this synthetic data, and cloning the workspaces that are required as input (that's done now in `test_GroupDiffractionCalibration`).  However, this adds a lot of complexity for very little gain, and makes things more difficult to maintain.  Here I'm adopting a compromise, and just regenerating the data.  As our test _instrument_ becomes more complicated, perhaps we should then change this.